### ISSUE TITLE
Discussion: Guests Eventually Stop Riding

### ISSUE THREAD

**Issue by Broxzier** [OpenRCT2/OpenRCT2#3216]
_Couldn't find it, but I remember seeing a similar issue about this a few months ago_

Making this issue, because I think it's a flaw in the game. This topic made me make this issue: https://openrct2.org/forums/topic/1074-guest-eventually-stop-riding/#comment-4836

I'm sure everyone who has played in one park for a couple of in-game years has probably noticed this at some points: guests loose interests in old rides to the point that they don't want to get in anymore. This makes upkeep costs too high, even when the ride was very successful before.

While there are several reasons why it is a good idea to make old rides less interesting (think about 'new' popular rides being the new star of a park; people revisiting after a long time, having lost interest in the ride they wanted to try once; Park is getting large, so they skip rides they have been in before) and the way RCT2 handles this has been implemented fairly okay.

What I do think is wrong however, is that after a long time, no one is willing to pay anything anymore to get into a ride. Personally I think it would be better to instead of having this as a penalty, making new rides have a popularity boost instead, which drops slowly, up to the point the boost is completely gone, leaving the ride with a minimal popularity factor, one that would still be enough for guests to enter a ride. In a real park where guests have to pay for rides, I don't think guests will stop wanting to pay for an old coaster. There are always guests that have not been in there before, or simply just like the ride and want to experience it again.

Maybe together with this system, guests can also keep track of how many times they went into a ride, making it slightly more boring every time. Then for new guests the ride still has the 'new' factor applied.

The way it is designed now just feels bad, it's too big of a penalty. At least that's my opinion, and seeing these thread confirms that I'm not the only one thinking so. I would like to hear your opinions on this.

**Comment by Gymnasiast** [comment]
I agree the system can be improved. There is also too much of a reliability penalty, IMO, making it hard to keep older rides (even those younger than 10 years) viable. In real life, rides often last several decades, with many parks keeping even older rides without much trouble.

Both are game mechanics of course, and therefore more of a challenge to right, as well as being more subject to discussion.

**Comment by jensj12** [comment]
What is currently the formula behind this?

**Comment by ed-foley** [comment]
The relevant parts are in `ride_ratings_calculate_value()` of [ride_ratings.c](https://github.com/OpenRCT2/OpenRCT2/blob/develop/src/ride/ride_ratings.c#L637).

Say a ride's nominal value based on its ratings is V.  Then the below is the ride's actual value as it ages.  (In practice, it will be slightly higher due to the way the integer arithmetic is performed.

0–4 months: V + 30
5–12 months: V + 10
13–39 months: V
40–63 months: 0.75 * V
64–87 months: 0.56 * V
88–103 months: 0.42 * V
104–119 months: 0.32 * V
120–127 months: 0.16 * V
128–199 months: 0.08 * V
200+ months: 0.56 * V  <-- **Is this a bug?**

I believe these are based on the RCT year (8 months per year).

In addition, the result of the above gets multiplied by 0.75 if there is another ride of the same type somewhere in the park.

If we were to amend this, I would probably remove the penalties for everything 40+ months old and instead treat the remaining adjustments as a "new ride bonus" rather than an "old ride penalty."  Besides, a 40-month-old ride means it has existed for 5 in-game years, which is longer than any built-in scenario.

**Comment by Broxzier** [comment]
@ed-foley Nice explanation!
The 200+ months one looks like a bug indeed, it doesn't make sense. The 128-199 month penalty seems too extreme to mee. That means that after 16 in-game years it has only 2/25th of its original ratings left. Surely a coaster of 16 years old is still exciting for people.

**Comment by jensj12** [comment]
What if the 200+ months penalty is not a bug but the others are? (looking at the code... it is a bug)
I'm in for removing the extra penalties after the first 87 months, not sure about the others yet, perhaps change the penalty from 25% to 20%? Even with a 25% penalty my best coasters still have an overflowing queue.

**Comment by tjwrona1992** [comment]
@ed-foley @Broxzier @jensj12 I believe the 200+ month change is not a bug and in fact is designed to give a boost to very old rides because once they become that old they are considered "vintage" rides. It was described in a YouTube video by Marcel Vos at some point but I don't remember exactly which video.

It makes sense though. In real life, REALLY old rides will eventually become a staple of a park and guests will come to the park just to ride the old ride that the park is known for.

**Comment by Wuzzy2** [comment]
Ouch. If this is true, then this completely kills the idea behind ride refurbishment. Because ride reburbishment resets the age.

**Comment by James103** [comment]
Would it be possible to add a drawback to ride refurbishment, so players can't just refurbish the ride all the time and have to actually build new rides?
* Every refurbishment increases the cost of the next refurbishment.
* Every refurbishment decreases the value of the ride.
* Every refurbishment decreases the max reliability of the ride.
* Refurbishing a ride takes time, during which the ride remains closed.
* Refurbishing a ride may fail, causing the ride to breakdown and possibly crash.
* The local authorities may prohibit rides from being refurbished.

Additionally, could you make it so the "easter egg" that increases the value of the ride can be disabled per scenario, keeping the ride value at/below 0.08x base value even if the ride is over 25 years old?

**Comment by ocalhoun6** [comment]
Any of the 'per scenario' options would require a new save format, yes?

**Comment by jensj12** [comment]
Ride refurbishment is not a new feature, just a shortcut. Even in RCT2, you can refurbish most rides by destroying and rebuilding them. The drawback of refurbishing is that is costs more money than rebuilding the ride. With the new save format, refurbishing will likely be further refined by splitting the build date and last refurbishment date, opening up more options to calculate the age penalty.

Anyways, this issue is about the ride value dropping very low over time (penalties of 84% and 92%). How about a bonus for guests that haven't ridden the ride before (or in the past 5 years)? That would keep the penalty, but still allow a small group of guests to ride and pay.

**Comment by tjwrona1992** [comment]
I would think that you would want to split build date and last refurbishment date into 2 separate things. Refurbishing a ride doesnt make it a whole new ride so it shouldnt reset the age so guests will pay complete full price again. It should only add a small bonus to what guests are willing to pay.
